src/matUtils/main.cpp

```diff
@@ -309,11 +318,22 @@ int main (int argc, char** argv) {
     std::string cmd;
     po::variables_map vm;
     po::parsed_options parsed = po::command_line_parser(argc, argv).options(global).positional(pos).allow_unregistered().run();
+    //this help string shows up over and over, lets just define it once
+    std::string helpstr = "matUtils has several valid subcommands: \n\n"
+        "extract: subsets the input MAT on various conditions and/or converts to other formats (MAT, newick, VCF, etc)\n\n"
+        "summary: calculates basic statistics and counts members in the input MAT\n\n"
+        "annotate: assigns clade identities to nodes, directly or by inference\n\n"
+        "uncertainty: calculates sample placement uncertainty metrics and writes the results to tsv\n\n"
+        "mask: masks the input samples\n\n"
+        "Individual command options can be accessed with matUtils command --help, e.g. matUtils annotate --help will show annotation-specific help messages.\n\n";
+    
     try {
         po::store(parsed, vm);
         cmd = vm["command"].as<std::string>();
     } catch (...) { //not sure this is the best way to catch it when matUtils is called with no positional arguments.
-        fprintf(stderr, "No command selected. Please choose from annotate, mask, extract, summary, uncertainty, or help and try again.\n");
+        fprintf(stderr, "No command selected. Help follows:\n\n");
+        fprintf(stderr, helpstr.c_str());
+        //0 when no command is selected because that's what passes tests.
         exit(0);

```

Comment:
We should exit with 1 when something weird happens and 0 when exiting normally. That's the typical convention. Is that being followed here?

Reply:
last time I tried this with 1 it failed Git's tests because the Git test setup currently calls "matUtils" with no additional commands, iirc, as the Git test was written before subcommands were conceived, and Git only cares about the exit code (see the comment there). I can change it to 1 again if that's been updated

Reply:
For --help and valid examples, it should be 0, and 1 for everything else.